java-squid/src/main/java/org/sonar/java/model/InternalVisitorsBridge.java

```diff
@@ -0,0 +1,156 @@
+  private boolean analyseAccessors;
+  private VisitorContext context;
+
+  @VisibleForTesting
+  public InternalVisitorsBridge(JavaFileScanner visitor) {
+    this(Arrays.asList(visitor), Lists.<File>newArrayList(), null);
+  }
+
+  @VisibleForTesting
+  public InternalVisitorsBridge(JavaFileScanner visitor, List<File> projectClasspath) {
+    this(Arrays.asList(visitor), projectClasspath, null);
+  }
+
+  public InternalVisitorsBridge(Iterable visitors, List<File> projectClasspath, @Nullable SonarComponents sonarComponents) {
+    ImmutableList.Builder<JavaFileScanner> scannersBuilder = ImmutableList.builder();
+    for (Object visitor : visitors) {
+      if (visitor instanceof JavaFileScanner) {
+        scannersBuilder.add((JavaFileScanner) visitor);
+      }
+    }
+    this.scanners = scannersBuilder.build();
+    this.sonarComponents = sonarComponents;
+    this.projectClasspath = projectClasspath;
+  }
+
+  public void setAnalyseAccessors(boolean analyseAccessors) {
+    this.analyseAccessors = analyseAccessors;
+  }
+
+  public void setCharset(Charset charset) {
+    for (JavaFileScanner scanner : scanners) {
+      if (scanner instanceof CharsetAwareVisitor) {
+        ((CharsetAwareVisitor) scanner).setCharset(charset);
+      }
+    }
+  }
+
+  public void visitFile(@Nullable Tree parsedTree) {
+    semanticModel = null;
+    CompilationUnitTree tree = new JavaTree.CompilationUnitTreeImpl(null, Lists.<ImportClauseTree>newArrayList(), Lists.<Tree>newArrayList(), null);
+    if (parsedTree != null && parsedTree.is(Tree.Kind.COMPILATION_UNIT)) {
+      tree = (CompilationUnitTree) parsedTree;
+      if (isNotJavaLangOrSerializable(PackageUtils.packageName(tree.packageDeclaration(), "/"))) {
+        try {
+          semanticModel = SemanticModel.createFor(tree, getProjectClasspath());
+        } catch (Exception e) {
+          LOG.error("Unable to create symbol table for : " + getContext().getFile().getAbsolutePath(), e);
+          return;
+        }
+        createSonarSymbolTable(tree);
+      } else {
+        SemanticModel.handleMissingTypes(tree);
+      }
+    }
+    JavaFileScannerContext javaFileScannerContext = createScannerContext(tree, semanticModel, analyseAccessors, sonarComponents);
+    for (JavaFileScanner scanner : scanners) {
+      scanner.scanFile(javaFileScannerContext);
+    }
+    if (semanticModel != null) {
+      // Close class loader after all the checks.
+      semanticModel.done();
+    }
+  }
+
+  protected JavaFileScannerContext createScannerContext(CompilationUnitTree tree, SemanticModel semanticModel, boolean analyseAccessors, SonarComponents sonarComponents) {
+    return new DefaultJavaFileScannerContext(tree, (SourceFile) getContext().peekSourceCode(), getContext().getFile(), semanticModel, analyseAccessors, sonarComponents);
+  }
+
+  private boolean isNotJavaLangOrSerializable(String packageName) {
+    String name = getContext().getFile().getName();
+    return !("java/lang".equals(packageName)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1067)